docs/source/model/signatures.rst

```diff
@@ -629,6 +629,167 @@ The same signature can be created explicitly as follows:
+    that MLflow accepts. 
+
+2. Log the langchain model with a valid input example:
+
+.. code-block:: python
+    
+    input_example = {
+        "messages": [
+            {
+                "role": "user",
+                "content": {
+                    "question": "What is the primary function of control rods in a nuclear reactor?",
+                    "answer": "To stir the primary coolant so that the neutrons are mixed well.",
+                },
+            },
+        ]
+    }
+
+    chain_path = "langchain_model.py"
+    with mlflow.start_run():
+        model_info = mlflow.langchain.log_model(
+            lc_model=chain_path, artifact_path="model", input_example=input_example
+        )
+
+At this stage, the input_example serves multiple purposes: it is used to infer the model signature, 
+validate that the model functions correctly when reloaded using the PyFunc flavor, and to generate 
+a corresponding serving_input_example. The generated signature is used to validate the model’s 
+functionality prior to deployment as well as providing guidance to callers of the model when 
+it is deployed if they are making requests with invalid data structures.
+For more information on the benefits and usage of the input example, refer to the :ref:`Model Input Example <input-example>` section.
+To learn how to validate the model locally before deployment, refer to the :ref:`Model Serving Payload Example <model-serving-payload-example>` section.
+
+If your model has an optional input field, you can use below input_example as a reference:
+
+.. code-block:: python
+
+    from mlflow.models import infer_signature
+
+    input_example = {
+        "messages": [
+            {
+                # specify name field in the first message
+                "name": "userA",
+                "role": "user",
+                "content": {
+                    "question": "What is the primary function of control rods in a nuclear reactor?",
+                    "answer": "To stir the primary coolant so that the neutrons are mixed well.",
+                },
+            },
+            {
+                # no name field in the second message, so `name` will be inferred as optional
+                "role": "user",
+                "content": {
+                    "question": "What is MLflow?",
+                    "answer": "MLflow is an open-source platform",
+                },
+            },
+        ]
+    }
+
+    print(infer_signature(input_example))
+
+3. Load the model back and make predictions:
+
+.. code-block:: python
+
+    loaded_model = mlflow.pyfunc.load_model(model_info.model_uri)
+    result = loaded_model.predict(input_example)
+
+    print(result)
+
+**A Dataclass object is also a valid way of submitting a signature. Passing a Dataclass
+definition to** `infer_signature` **will generate the corresponding model signature equivalent**:
```

Comment:
```suggestion
instance to** `infer_signature` **will generate the corresponding model signature equivalent**:
```

`Dataclass` definition sounds like this:

```
@dataclass
class Foo:
    pass

infer_signature(Foo) # not infer_signature(Foo)
```suggestion
definition to** `infer_signature` **will generate the corresponding model signature equivalent**:
```
```